src/common/common.h

```diff
@@ -869,8 +882,252 @@ void GDS_breakpoint(int);
 
 /* The default lseek offset type.  Changed from nothing to (off_t) to correctly support 64 bit IO */
 #ifndef LSEEK_OFFSET_CAST
+#ifdef LSB_BUILD
+#define LSEEK_OFFSET_CAST (loff_t)
+#else
 #define LSEEK_OFFSET_CAST (off_t)
 #endif
+#endif
+
+// LSB uses 32bit IO functions by default
+#ifdef LSB_BUILD
+#define off_t loff_t
+#define fpos_t fpos64_t
+#define dirent dirent64
+#define rlimit rlimit64
+#define STAT stat64
+#define FSTAT fstat64
+#define LSTAT lstat64
+#define FLOCK flock64
+#else
+#define STAT stat
+#define FSTAT fstat
+#define LSTAT lstat
+#define LSTAT lstat
+#define FLOCK flock
+#endif
+
+#ifdef WIN_NT
+#define mode_t int
+#endif
+
+namespace fb_io {
+
+inline int open(const char* pathname, int flags, mode_t mode = 0666) {
+	int rc;
+	do {
+#ifdef LSB_BUILD
+		rc = open64(pathname, flags, mode);
+#else
+		rc = ::open(pathname, flags, mode);
+#endif
+        } while (rc == -1 && SYSCALL_INTERRUPTED(errno));
+	return rc;
```

Comment:
OK, I will fix it.

Dmitry Starodubov

25.05.2016 15:19, Adriano dos Santos Fernandes пишет:
>
> In src/common/common.h 
> <https://github.com/FirebirdSQL/firebird/pull/28#discussion_r64561674>:
>
> > +#ifdef WIN_NT
> > +#define mode_t int
> > +#endif
> > +
> > +namespace fb_io {
> > +
> > +inline int open(const char* pathname, int flags, mode_t mode = 0666) {
> > +	int rc;
> > +	do {
> > +#ifdef LSB_BUILD
> > +		rc = open64(pathname, flags, mode);
> > +#else
> > +		rc = ::open(pathname, flags, mode);
> > +#endif
> > +        } while (rc == -1 && SYSCALL_INTERRUPTED(errno));
> > +	return rc;
>
> Indentation of all "} while" in this file is incorrectly done with 
> spaces intead of tabs.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/FirebirdSQL/firebird/pull/28/files/39c19e5c8eae58030ce5d8c1ae9b402a48e4c516#r64561674>
>